assets/js/blocks/price-filter/block.js

```diff
@@ -38,33 +42,131 @@ const PriceFilterBlock = ( { attributes, isPreview = false } ) => {
 		},
 	} );
 
-	const { showInputFields, showFilterButton } = attributes;
-	const minConstraint = isLoading
-		? undefined
-		: // Round up to nearest 10 to match the step attribute.
-		  Math.floor( parseInt( results.min_price, 10 ) / 10 ) * 10;
-	const maxConstraint = isLoading
-		? undefined
-		: // Round down to nearest 10 to match the step attribute.
-		  Math.ceil( parseInt( results.max_price, 10 ) / 10 ) * 10;
+	const [ minPrice, setMinPrice ] = useState();
+	const [ maxPrice, setMaxPrice ] = useState();
+	const [ minConstraint, setMinConstraint ] = useState();
+	const [ maxConstraint, setMaxConstraint ] = useState();
+	const prevMinConstraint = usePrevious( minConstraint );
+	const prevMaxConstraint = usePrevious( maxConstraint );
 
+	/**
+	 * Updates the query after a short delay.
+	 */
+	const [ debouncedUpdateQuery ] = useDebouncedCallback( () => {
+		onSubmit();
+	}, 500 );
+
+	/**
+	 * Updates the query based on slider values.
+	 */
+	const onSubmit = useCallback( () => {
+		setMinPriceQuery( minPrice === minConstraint ? undefined : minPrice );
+		setMaxPriceQuery( maxPrice === maxConstraint ? undefined : maxPrice );
+	}, [ minPrice, maxPrice, minConstraint, maxConstraint ] );
+
+	/**
+	 * Callback when slider is changed.
+	 */
 	const onChange = useCallback(
 		( prices ) => {
-			if ( prices[ 0 ] === minConstraint ) {
-				setMinPrice( undefined );
-			} else if ( prices[ 0 ] !== minPrice ) {
+			if ( prices[ 0 ] !== minPrice ) {
 				setMinPrice( prices[ 0 ] );
 			}
-
-			if ( prices[ 1 ] === maxConstraint ) {
-				setMaxPrice( undefined );
-			} else if ( prices[ 1 ] !== maxPrice ) {
+			if ( prices[ 1 ] !== maxPrice ) {
 				setMaxPrice( prices[ 1 ] );
 			}
 		},
 		[ minConstraint, maxConstraint, minPrice, maxPrice ]
 	);
 
+	/**
+	 * Track price STATE changes - if state changes, update the query.
+	 */
```

Comment:
Inline comments inside logic blocks are not multi-line (same for other places in this file). [Reference](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#comments). Btw, I _love_ the descriptions of what is happening - very useful to our future selves or others looking at the code 👏